doc/track_2_migration_guidance.md

```diff
@@ -0,0 +1,116 @@
+# Track 2 Migration Guidance
+
+Azure CLI is built upon Azure Python SDK. Recently Azure Python SDK announced next generation product. It is named Track 2 SDK. The old version of SDK is named Track 1. It claims that it has great advantages than Track 1 SDK. It is not compatible with Track 1 SDK. Azure CLI developers need to spend considerable time and do some work to migrate from Track 1 SDK to Track 2 SDK. Let's see an example of Track 2 SDK. [azure-mgmt-compute 17.0.0b1](https://pypi.org/project/azure-mgmt-compute/17.0.0b1/) introduces important breaking changes and important new features like unified authentication and asynchronous programming.
+
+This document summarizes typical issues and solutions when adopting Track 2 SDK in Azure CLI.
+
+Example PRs:
+1. [Compute PR](https://github.com/Azure/azure-cli/pull/15750)
+2. [Network PR](https://github.com/Azure/azure-cli/pull/16245)
+
+Below is a list of typical issues.
+
+1. Long running operation function name change
+2. Property name change
+3. Class name change
+4. Error type change
+5. No enum type
+6. Class hierarchy change
+7. Obtaining Subscription
+8. Multi-API support
+9. Fixing mocked object
+10. Modifying patch_models.py to include missing packages
+11. Missing x-ms-authorization-auxiliary
+
+**Long running operation function name change**
+
+Long running operations have changed their function names in Track 2 SDK. A `begin_` prefix is added. E.g. `create_or_update` becomes `begin_create_or_update`. `delete` becomes `begin_delete`. It is a naming convention in Track 2 SDK to indicate that an operation is a long running operation.
+
+**Property name change**
+
+Some of property names change in Track 2 SDK.
+
+Examples:
+
+```
+hyperVgeneration -> hyperVGeneration
+disk_mbps_read_only -> disk_m_bps_read_only
+disk_mbps_read_write -> disk_m_bps_read_write
+virtual_machine_extension_type -> type_properties_type
+type1 -> type_properties_type
+instance_ids -> vm_instance_i_ds
+diskMbpsReadWrite -> diskMBpsReadWrite
+```
+
+Some changes are unnecessary, even wrong in English. I opened an [issue](https://github.com/Azure/autorest.python/issues/850) to track this problem.
```

Comment:
It's better to use the GitHub convention to represent an issue. `https://github.com/Azure/autorest.python/issues/850` will be converted to https://github.com/Azure/autorest.python/issues/850.

```suggestion
Some changes are unnecessary, even wrong in English. I opened [Azure/autorest.python#850](https://github.com/Azure/autorest.python/issues/850) to track this problem.
```